x-pack/plugins/security_solution/common/endpoint/service/trusted_apps/validations.ts

```diff
@@ -57,10 +50,18 @@ export const hasSimpleExecutableName = ({
   type: TrustedAppEntryTypes;
   value: string;
 }): boolean => {
-  if (type === 'wildcard') {
-    return os === OperatingSystem.WINDOWS ? WIN_EXEC_PATH.test(value) : UNIX_EXEC_PATH.test(value);
+  if (type !== 'wildcard') {
+    return true;
   }
-  return true;
+
+  const separator = os === OperatingSystem.WINDOWS ? '\\' : '/';
+  const lastString = value.split(separator).pop();
+
+  if (!lastString) {
+    return false;
+  }
+
+  return /[\*\?]/.test(lastString) === false;
```

Comment:
Could we use https://github.com/uhop/node-re2 here instead of the built-in `RegExp` library to further protect ourselves?

Reply:
Hi @kobelb - thanks for the comment. The change here (almost all of it) is actually a copy of the code in `main` - we had already altered this method in `main` for 8.2 and we know it works. Do you know if `node-re2` can be used both in browser and server side? this utility method is used in both.

Reply:
@kobelb - we removed the RegEx here entirely for this `7.17.4` patch which goes out soon.  We'll look into `node-re2` when we've got a bit more time.

Related change: https://github.com/elastic/kibana/pull/131085/commits/0dc084b3eb2a1029ac32a66e49ac4d45f17649f7